### ISSUE TITLE
linux sessions can not get into shell

### ISSUE THREAD

**Issue by z0sen** [BishopFox/sliver#1259]
After input shell command it's just output like this:

<img width="931" alt="image" src="https://github.com/BishopFox/sliver/assets/49097104/6fd30dd2-b090-4130-84c2-788fc7d1a150">

just didn't come out an interactive shell, and it's not working with ctrl+c ctrl+z or ctrl+d, i have to shut down my terminal and restart the sliver so that i could use sliver again

**Comment by moloch--** [comment]
Do you have any more information about the remote system? Is it an odd distribution/container? Not enough information to help.

**Comment by z0sen** [comment]
surely it's not an odd distribution/container, cause I was able to use it normally a few days ago, maybe after i installed armory or maybe i restart my mac, then i could not get into shell, here is the info below:
<img width="1006" alt="image" src="https://github.com/BishopFox/sliver/assets/49097104/769b9836-a178-4ca8-b46b-0d71e0c62fd7">

**Comment by moloch--** [comment]
Does `/bin/bash` exist on the remote system? We try to fallback to `/bin/sh` if it doesn't exist, but the fallback is maybe buggy?

**Comment by z0sen** [comment]
> Does `/bin/bash` exist on the remote system? We try to fallback to `/bin/sh` if it doesn't exist, but the fallback is maybe buggy?

that's surely exists, and i could use msf's shell command to get into the interactive shell:
<img width="608" alt="image" src="https://github.com/BishopFox/sliver/assets/49097104/1ccdfe0d-41dc-4d81-90da-dc6415533adc">

**Comment by moloch--** [comment]
If possible can you re-test with an implant compiled with `--debug`?

**Comment by z0sen** [comment]
> If possible can you re-test with an implant compiled with `--debug`?

it's really strange, everything worked fine after I restarted the server just now, and i restarted the mtls jobs, then i got the new sessions, everything just worked fine then.

**Comment by moloch--** [comment]
Okay, sounds like it could be a race condition or dead lock in our tunnel code.

**Comment by z0sen** [comment]
> Okay, sounds like it could be a race condition or dead lock in our tunnel code.

yeah, i think so, is there any logs i could give u (i don't konw if there some log path and where)? then u might be able to fix that with those logs

**Comment by moloch--** [comment]
The server log may contain errors `~/.sliver/logs/sliver.log` but the default log level is `info` so it may be incomplete depending on the nature of the failure; probably worth checking to see if there's anything in there though.

**Comment by z0sen** [comment]
> The server log may contain errors `~/.sliver/logs/sliver.log` but the default log level is `info` so it may be incomplete depending on the nature of the failure; probably worth checking to see if there's anything in there though.

the log level was set to 5 by me when i first time started server, there are some logs but i don't is that useful , maybe i could sent the log file to u, through telegram or protonmail?

**Comment by moloch--** [comment]
Not necessary, the tunnel code needs a fresh coat of paint to address issues like this. Please let us know if you find a reliable method to produce the behavior you observed.

**Comment by z0sen** [comment]
Okay, I'll keep an eye out.

**Comment by 1n7erface** [comment]
[https://github.com/BishopFox/sliver/issues/1146](url) 、I have also encountered this issue, but restarting can solve it

**Comment by z0sen** [comment]
> [https://github.com/BishopFox/sliver/issues/1146](url) 、I have also encountered this issue, but restarting can solve it

my server linux too, but my client is intel mac not m1

**Comment by z0sen** [comment]
After entering a shell, there is a chance that typing "exit" will fail to exit the shell. It won't roll back to the USE session and, after directly closing the terminal, running "sliver-client" again won't allow re-entering the shell. Trying to input the shell command again will result in being stuck, just like the previous state I mentioned.

**Comment by moloch--** [comment]
I think these issues are addressed in recent changes to `shell`
